### PR TITLE
Adds example for files to parse tsv files

### PR DESCRIPTION
**Pull request by sagar-shah94**

### CODE REVIEW

File: file/index.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    }).catch(function (err) {
+        callback(err); // Return when some error occurred while listing objects in bucket
+        console.log("Error in listing bucket", JSON.stringify(err) + "\n");
+    });
+
+    // Waiting for listing of objects so we can get all modified files
+    await listObjectsPromise;
+
+    // Sort in ascending order
+    modifiedFiles.sort(function (a, b) { return (a.LastModified > b.LastModified) ? 1 : ((b.LastModified > a.LastModified) ? -1 : 0); });
```
Comment (tabalinas) [inline file/index.js]: use lambdas `(dt1, dt2) => dt1 - d2`

File: file/index.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+    // Process files one by one
+    for (var index = 0; index < modifiedFiles.length; index++) {
```
Comment (tabalinas) [inline file/index.js]: modifiedFiles.forEach?

File: file/index.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        let params = {
+            Bucket: bucketName,
+            Key: modifiedFile.Key
+        };
+
+        let getObjectPromise = s3.getObject(params).promise();
+
+        getObjectPromise.then(function (result) {
+            let fileData = result.Body.toString('utf-8');
+            var rows=fileData.split("\n");
```
Comment (tabalinas) [inline file/index.js]: use consistent code formatting (add spaces around operators), not only for this line, do this everywhere in the code

File: file/index.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+async function update(state, secrets, callback) {
+    let response = initializeResponse(state);
+
+    let modifiedFiles = [];
+
+    let listObjectsPromise = s3.listObjects(bucketParams).promise();
+
+    listObjectsPromise.then(function (result) {
+        for (var index = 0; index < result.Contents.length; index++) {
```
Comment (tabalinas) [inline file/index.js]: use `let` (everywhere inside functions)

File: file/index.js [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    // Process files one by one
+    for (var index = 0; index < modifiedFiles.length; index++) {
+        let modifiedFile = modifiedFiles[index];
+
+        let params = {
+            Bucket: bucketName,
+            Key: modifiedFile.Key
+        };
+
+        let getObjectPromise = s3.getObject(params).promise();
```
Comment (tabalinas) [inline file/index.js]:
```js
await  s3.getObject(params).then((result => {
 ...
});
```

File: file/index.js [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+                for(var j in cols){
+                    obj[headers[j]]=cols[j];
+                }
+                if (modifiedFile.Key.endsWith("_delete.tsv")) {
+                    response.delete.near_earth_objects.push(obj);
+                } else{
+                    response.insert.near_earth_objects.push(obj);
+                }
+            }
+        }).catch(function (err) {
```
Comment (tabalinas) [inline file/index.js]: use lambda

File: file/index.js [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+};
+
+async function update(state, secrets, callback) {
+    let response = initializeResponse(state);
+
+    let modifiedFiles = [];
+
+    let listObjectsPromise = s3.listObjects(bucketParams).promise();
+
+    listObjectsPromise.then(function (result) {
```
Comment (tabalinas) [inline file/index.js]: use lambda

File: file/index.js [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+
+    // Once response in generated use callback to finish lambda execution with response
+    callback(null, response);
+}
+
+function initializeResponse(state) {
+    // Don't assign the value directly, it means you are assigning as a reference, 
+    // Now state of response doesn't get affected when change state variable in method
+    return {
+        "state": {...state},
```
Comment (tabalinas) [inline file/index.js]: it's weird way to declare js objects, usually you don't put field names in quotes. While you will have to do that for `delete` I believe, it's a reserved word (it's sad that we used this name in the first place).

File: file/index.js [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+    let response = initializeResponse(state);
+
+    let modifiedFiles = [];
+
+    let listObjectsPromise = s3.listObjects(bucketParams).promise();
+
+    listObjectsPromise.then(function (result) {
+        for (var index = 0; index < result.Contents.length; index++) {
+            
+            // Only JSON files to be processed
```
[the remark this answers is not in the document]
Comment (sagar-shah94) [reply file/index.js]: my bad!

File: file/index.js [hunk 10]
```diff
@@ -1,120 +1,120 @@
...
+    onError(err); // Return when some error occurred while listing objects in bucket
+    console.log("Error in listing bucket", JSON.stringify(err) + "\n");
+  });
+
+  // Waiting for listing of objects so we can get all modified files
+  await listObjectsPromise;
+
+  // Sort in ascending order
+  modifiedFiles.sort((a, b) => {
+    return a - b;
```
Comment (tabalinas) [inline file/index.js]: wait, we are comparing `LastModified`, not objects?!

File: file/index.js [hunk 11]
```diff
@@ -1,120 +1,120 @@
...
+let bucketParams = {
+  Bucket: bucketName
+};
 
-var limitNumberOfFileToUpdateInOneRun = 2;
+let limitNumberOfFileToUpdateInOneRun = 2;
 
 exports.handler = (request, context, callback) => {
-    update(request.state, request.secrets, callback);
+  update(request.state, request.secrets, callback);
```
Comment (tabalinas) [inline file/index.js]: rename onError here too (https://github.com/fivetran/functions/pull/1#discussion_r194258157)?

File: file/index.js [hunk 12]
```diff
@@ -1,120 +1,120 @@
...
-    };
-}
\ No newline at end of file
+  // Don't assign the value directly, it means you are assigning as a reference, 
+  // Now state of response doesn't get affected when change state variable in method
+  return {
+    state: { ...state
+    },
+    insert: {
+      "near_earth_objects": []
```
Comment (tabalinas) [inline file/index.js]: same for these field names (remove quotes), we need it only for `delete` as reserved word.

File: file/index.js (hunk 10 above)
Comment (sagar-shah94) [reply file/index.js]: I changed code to sort based on `LastModified` https://github.com/fivetran/functions/pull/1/commits/0596a91870f5e84ca56b1c7a3791be49260477a9

**Pull request merged**